pkg/minishift/provisioner/minishift_provisioner.go

```diff
@@ -88,6 +88,41 @@ func (provisioner *MinishiftProvisioner) Package(name string, action pkgaction.P
 	return errors.New("Custom provisioner does not support package")
 }
 
+func (provisioner *MinishiftProvisioner) remoteHostPrepration() error {
+	packageList := map[string]string{
+		"firewalld": "firewall-cmd",
+		"docker":    "docker",
+		"net-tools": "netstat"}
+
+	// Install required packages if not present.
+	for pkg, cmd := range packageList {
+		if out, err := provisioner.SSHCommand(fmt.Sprintf("which %s || sudo yum install -y %s", cmd, pkg)); err != nil {
+			return fmt.Errorf("Error installing package %s: %s", pkg, out)
+		}
+	}
+
+	// 	Start the firewalld service.
+	if _, err := provisioner.SSHCommand("sudo systemctl start firewalld"); err != nil {
+		return fmt.Errorf("Error starting firewalld service: %s", err)
+	}
+
+	firewallCommandsToExecute := []string{
+		"sudo firewall-cmd --permanent --add-port 2376/tcp --add-port 8443/tcp --add-port 80/tcp",
```

Comment:
not so happy with all these commands being everywhere in the file... makes maintenance very hard